Add num workers cli arg

chadykamar opened the pull request:
<!-- This line specifies which issue to close after the pull request is merged. -->
Fixes #6316

LysandreJik reviewed (approved):
LGTM

sgugger commented on src/transformers/training_args.py:
```diff
@@ -122,6 +122,8 @@ class TrainingArguments:
             Number of update steps between two evaluations if :obj:`evaluation_strategy="steps"`. Will default to the
             same value as :obj:`logging_steps` if not set.
+        dataloader_num_workers (:obj:`int`, `optional`, defaults to 0):
+            Number of subprocesses to use for data loading. 0 means that the data will be loaded in the main process.
```
```suggestion
            Number of subprocesses to use for data loading (PyTorch only). 0 means that the data will be loaded in the main process.
```

sgugger commented on src/transformers/training_args.py:
```diff
@@ -259,6 +261,10 @@ class TrainingArguments:
     eval_steps: int = field(default=None, metadata={"help": "Run an evaluation every X steps."})
+    dataloader_num_workers: int = field(
+        default=0,
+        metadata={"help": "Number of subprocesses to use for data loading. 0 means that the data will be loaded in the main process."}
```
```suggestion
        metadata={"help": "Number of subprocesses to use for data loading (PyTorch only). 0 means that the data will be loaded in the main process."}
```

sgugger reviewed (approved):
Thanks for taking care of this. I just have two nits since the `TrainingArguments` are (for now) subclassed by TF so we need to specify it's just for PyTorch.

jplu reviewed (approved):
LGTM!

Just a tiny comment, this parameter might be useful for TF as well when doing some dataset pre-processing for the `num_parallel_calls` parameter of few functions such as [map](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#map) or [interleave](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#interleave)

chadykamar commented:
I've addressed the comments regarding the docstring and help message. 

I'm a little less familiar with TensorFlow 2.0, but it seems like any preprocessing is done by the user before passing `train_dataset` and `eval_dataset` to the `TFTrainer` so there isn't an opportunity to set `num_parallel_calls` (I wasn't able to find any calls to `map` or `interleave` save for some markdown examples).

jplu commented:
Indeed there is no call directly in the trainer and these functions have to be run directly in the example script. Nevertheless, this parameter is still useful as it can be used directly in the example script, instead of updating it manually.
